packages/ra-ui-materialui/src/button/BulkDeleteButton.tsx

```diff
@@ -31,12 +30,15 @@ import BulkDeleteWithUndoButton, {
  *     </List>
  * );
  */
-const BulkDeleteButton: FC<BulkDeleteButtonProps> = ({ undoable, ...props }) =>
-    undoable ? (
-        <BulkDeleteWithUndoButton {...props} />
+const BulkDeleteButton = (props: BulkDeleteButtonProps) => {
+    const { undoable, ...rest } = props;
+
+    return undoable ? (
+        <BulkDeleteWithUndoButton {...rest} />
     ) : (
-        <BulkDeleteWithConfirmButton {...props} />
+        <BulkDeleteWithConfirmButton {...rest} />
```

Comment:
Nope, we should actually pass `props` here, as you can have a button which will ask for confirmation and still display an undoable notification. See https://github.com/marmelab/react-admin/pull/6472/files#diff-cf6d8ea9685fb8b58acfbc7f9afd4e47f13fd410266dcdb1c5f3bd642d247ecdR117

Reply:
This is something we'll probably revisit later though